ClientApp/viking-vault/src/display_user_list/UserData.tsx

```diff
@@ -24,59 +40,123 @@ const defaultUser = {
     email: "no-data",
     pictureLink: "",
     cardNumber: "no-data",
-    expirationDate: "no-data"
+    expirationDate: "no-data",
+    openDeleteUserModal: false
 };
 
-class UserData extends React.Component<IUserData, IUserData>{
+class UserData extends React.Component<IUserDataProp, IPageState>{
     
-    constructor(props: IUserData)
+    constructor(props: IUserDataProp)
     {
         super(props);
 
         if(props !== null)
         {
             this.state = {
-                id: props.id,
-                firstName: props.firstName,
-                lastName: props.lastName,
-                address: props.address,
-                email: props.email,
-                pictureLink: props.pictureLink,
-                cardNumber: props.cardNumber,
-                expirationDate: props.expirationDate
+                    user : {
+                    id: props.user.id,
+                    firstName: props.user.firstName,
+                    lastName: props.user.lastName,
+                    address: props.user.address,
+                    email: props.user.email,
+                    pictureLink: props.user.pictureLink,
+                    cardNumber: props.user.cardNumber,
+                    expirationDate: props.user.expirationDate
+                },
+                openDeleteUserModal: false,
```

Comment:
If you have to prefix multiple properties with the same name, it's usually a good sign that they belong nicely together :)